docker/Dockerfile

```diff
@@ -108,6 +108,12 @@ RUN mkdir /rust /cargo
 
 RUN curl -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain stable
 
+
+# arm64 builds consume a lot of memory if `CARGO_NET_GIT_FETCH_WITH_CLI` is not
+# set to true, so we expose it as a build-arg.
+ARG CARGO_NET_GIT_FETCH_WITH_CLI=false
+ENV CARGO_NET_GIT_FETCH_WITH_CLI=$CARGO_NET_GIT_FETCH_WITH_CLI
```

Comment:
er, note that `ENV` defines a variable that will be present in containers started from the final image, which I don't think is really what you want here.

(I guess it won't do any harm, but it does seem like an odd variable for us to export to our image).

Instead you have to do `CARGO_NET_GIT_FETCH_WITH_CLI=$CARGO_NET_GIT_FETCH_WITH_CLI` before the `pip` that needs it.

Reply:
Yeah, though `ENV` should only affect the current build stage and so not be in the final image.

Reply:
ah yes, fair